src/core/client/stream/tabs/Comments/Stream/Announcement/AnnouncementContainer.tsx

```diff
@@ -0,0 +1,69 @@
+import React, {
+  FunctionComponent,
+  useCallback,
+  useEffect,
+  useState,
+} from "react";
+import { graphql } from "react-relay";
+
+import { ANNOUNEMENT_DISMISSED_KEY } from "coral-framework/constants";
+import { useCoralContext } from "coral-framework/lib/bootstrap";
+import { withFragmentContainer } from "coral-framework/lib/relay";
+
+import { AnnouncementContainer_settings as SettingsData } from "coral-stream/__generated__/AnnouncementContainer_settings.graphql";
+
+import Announcement from "./Announcement";
+
+interface Props {
+  settings: SettingsData;
+}
+
+export const AnnouncementContainer: FunctionComponent<Props> = ({
+  settings,
+}) => {
+  const { localStorage } = useCoralContext();
+  const [dismissed, setDismissed] = useState(false);
+  useEffect(() => {
+    async function getDismissedStatus() {
+      if (!settings.announcement) {
+        setDismissed(true);
```

Comment:
You can just ignore changing state when `settings.announcement` is not set right, because it's rendered as `null` anyway.